src/views/preview/project-view.jsx

```diff
@@ -246,19 +251,33 @@ class Preview extends React.Component {
                             });
                         }
                         const sprites = projectData[0].targets.length - 1; // don't count stage
-                        const scripts = projectData[0].targets
-                            .map(target =>
-                                Object.values(target.blocks)
-                                    .filter(block => block.topLevel).length
-                            )
-                            .reduce((accumulator, currentVal) => accumulator + currentVal, 0);
-                        this.setState({
+                        const blocks = projectData[0].targets
+                            .map(target => Object.values(target.blocks))
+                            .reduce((accumulator, currentVal) => accumulator.concat(currentVal), []);
+
+                        const scriptCount = blocks.filter(block => block.topLevel).length;
+
+                        const newState = {
                             extensions: Array.from(extensionSet),
                             modInfo: {
-                                scripts: scripts,
+                                scripts: scriptCount,
                                 sprites: sprites
                             }
-                        });
+                        };
+
+                        if (showAlerts) {
+                            // Check for username block only if user is logged in
+                            if (this.props.isLoggedIn) {
+                                newState.showUsernameBlockAlert = blocks.some(block =>
+                                    block.opcode === 'sensing_username');
+                            } else { // Check for cloud vars only if user is logged out
+                                const stage = projectData[0].targets[0];
+                                newState.showCloudDataAlertObject = Object.values(stage.variables)
+                                    .some(variable => variable.length === 3); // 3 entries if cloud var
```

Comment:
cc / @chrisgarrity, just wanted to make sure you knew about these changes, and also that we will need to set these flags when parsing 2.0 projects as well